dali/pipeline/operator/op_schema.h

```diff
@@ -454,6 +460,44 @@ class DLL_PUBLIC OpSchema {
     return *this;
   }
 
+  /**
+   * @brief Marks an argument as deprecated in favor of a new argument
+   * @remarks There are three ways to deprecate an argument
+   *          1. renamed_to provided, means the argument has been renamed and we can safely
+   *              propagate the value to the new argument name.
+   *          2. renamed_to not provided and ignore==true, means the operator will not use the
+   *              argument at all and it can be safely discarded.
+   *          3. renamed_to not provided and ignore==false, means the operator will still use the
+   *              deprecated argument until it is finally removed completely from the schema.
+   */
+  DLL_PUBLIC inline OpSchema &DeprecateArgInFavorOf(const std::string& arg_name,
+                                                    std::string renamed_to,
+                                                    std::string msg = {}) {
+    if (msg.empty())
+      msg = DefaultDeprecatedArgMsg(arg_name);
+    deprecated_arguments_[arg_name] = {std::move(msg), std::move(renamed_to), false};
+    return *this;
+  }
+
+  /**
+   * @brief Marks an argument as deprecated in favor of a new argument
+   * @remarks There are three ways to deprecate an argument
+   *          1. renamed_to provided, means the argument has been renamed and we can safely
+   *              propagate the value to the new argument name.
+   *          2. renamed_to not provided and ignore==true, means the operator will not use the
+   *              argument at all and it can be safely discarded.
+   *          3. renamed_to not provided and ignore==false, means the operator will still use the
+   *              deprecated argument until it is finally removed completely from the schema.
+   */
```

Comment:
```suggestion
  /**
   * @brief Marks an argument as deprecated
   * @remarks There are three ways to deprecate an argument
   *          1. removed==true, means the operator will not use the
   *              argument at all and it can be safely discarded.
   *          2. removed==false, means the operator will still use the
   *              deprecated argument until it is finally removed completely from the schema.
   *          3. For renaming the argument see DeprecateArgInFavorOf
   */
```